lib/SILGen/SILGenExpr.cpp

```diff
@@ -4438,10 +4540,14 @@ void SILGenFunction::emitBindOptionalAddress(SILLocation loc,
   // Since we know that we have an address, we do not need to worry about
   // ownership invariants. Instead just use a select_enum_addr.
   SILBasicBlock *someBB = createBasicBlock();
+  if (mayShortCut) {
+    B.createBranch(loc, someBB);
```

Comment:
You don’t need the basic block at all in this case I think. It should be enough to just bitcast the optional to its payload type.